src/components/AssetTree/AssetTree.tsx

```diff
@@ -159,10 +161,11 @@ class AssetTree extends React.Component<AssetTreeProps, AssetTreeState> {
     }
   };
 
-  onSelectNode = (returnAsset: OnSelectAssetTreeParams) => {
+  onSelectNode = (selectedNode: AssetTreeNode) => {
     const { onSelect } = this.props;
+    const { asset, key, isLeaf } = selectedNode;
     if (onSelect) {
-      onSelect(returnAsset);
+      onSelect({ node: asset, key, isLeaf, title: this.getDisplayName(asset) });
```

Comment:
Can't the shape of `onSelect` be `AssetTreeNode` -- can't we unify here `node` and `asset`?

Reply:
@valerii-cognite you mean to rename AssetTreeNode  `asset: Asset;` property to `node: Asset`?

Reply:
yes, or the other way around, currently it seems a bit odd having two similar structures with only one difference in the key name

Reply:
hmm they are a bit different
```
  isLeaf?: boolean;
  node?: Asset;
```
are optional on OnSelectAssetTreeParams for some reason